tools/panda.py

```diff
@@ -0,0 +1,140 @@
+
+
+async def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-e",
+                        "--executable",
+                        type=pathlib.Path,
+                        help="path to redpanda executable",
+                        default="redpanda")
+    parser.add_argument("--nodes", type=int, help="number of nodes", default=3)
+    parser.add_argument("-d",
+                        "--directory",
+                        type=pathlib.Path,
+                        help="data directory",
+                        default="data")
+    parser.add_argument("--rpc-port", type=int, help="rpc port", default=33145)
+    parser.add_argument("--kafka-port",
+                        type=int,
+                        help="kafka port",
+                        default=9092)
+    parser.add_argument("--admin-port",
+                        type=int,
+                        help="admin port",
+                        default=9644)
+    parser.add_argument("--address",
+                        type=str,
+                        help="listening address",
+                        default="0.0.0.0")
+    args = parser.parse_args()
+
+    seed_servers = [
+        NetworkAddress(args.address, args.rpc_port + i)
+        for i in range(args.nodes)
+    ]
+
+    def make_node_config(i, data_dir):
+        make_address = lambda p: NetworkAddress(args.address, p + i)
+        rpc_address = seed_servers[i]
+        redpanda = RedpandaConfig(data_directory=data_dir,
+                                  rpc_server=rpc_address,
+                                  advertised_rpc_api=rpc_address,
+                                  kafka_api=make_address(args.kafka_port),
+                                  admin=make_address(args.admin_port),
+                                  seed_servers=seed_servers,
+                                  empty_seed_starts_cluster=False)
+        return NodeConfig(redpanda=redpanda)
+
+    def pathlib_path_representer(dumper, path):
+        return dumper.represent_scalar("!Path", str(path))
+
+    def get_config_dumper():
+        d = yaml.SafeDumper
+        d.add_representer(pathlib.PosixPath, pathlib_path_representer)
+        return d
+
+    def prepare_node(i):
+        node_dir = args.directory / f"node{i}"
+        data_dir = node_dir / "data"
+        conf_file = node_dir / "config.yaml"
+
+        node_dir.mkdir(parents=True, exist_ok=True)
+        data_dir.mkdir(parents=True, exist_ok=True)
+
+        config = make_node_config(i, data_dir)
+        with open(conf_file, "w") as f:
+            yaml.dump(dataclasses.asdict(config),
+                      f,
+                      indent=2,
+                      Dumper=get_config_dumper())
+        return conf_file
+
+    configs = [prepare_node(i) for i in range(args.nodes)]
+    nodes = [Redpanda(args.executable, c) for c in configs]
+    await asyncio.wait(map(lambda r: asyncio.create_task(r.run()), nodes))
```

Comment:
nit: I think it'd be helpful to print some copy-pastable commonly-used addresses/ports based on the values generated, like:
- admin endpoint port for each node
- list of all kafka endpoints (for use in applications)

Reply:
fixed. now if you type something like `*h*<ENTER>` the program will print

```
admin endpoint: http://localhost:9644
admin endpoint: http://localhost:9645
admin endpoint: http://localhost:9646
```